test/test_helper.rb

```diff
@@ -8,4 +8,5 @@
 
 Procore.configure do |config|
   config.host = "https://procore.example.com"
+  config.default_version = "v1.0"
```

Comment:
So without this, the test suite is failing. By nature of it being a default, it shouldn't need to be explicitly set. `v1.0` has been added to `lib/procore/defaults.rb` - that is great, it just isn't being used. In https://github.com/procore/ruby-sdk/blob/1b577da589e1af8bf3fa8b3fca85f3e91070cbce/lib/procore/configuration.rb#L105-L112 

You will need to set it - that way the default is actually set.

Reply:
I think the variable name could be little confusing. The config, default_version, should be set by the consumer of the SDK. We don't want to make any assumptions about what is the default_version for each user of the SDK.

Here are some other possible names. 

* `root_version` 
* `base_version`

Reply:
> Setting a default_version is now required during configuration, and should be either v1.0 (recommended) or vapid (legacy).

As Gil said, we require the consumer to configure `default_version` which is used if `version:` is not provided in the request. So it does need to be set in the test helper, and we raise an error if it's not defined:
https://github.com/procore/ruby-sdk/blob/efd89478f1838ac0ff992785b5550be048e2cc06/lib/procore/configuration.rb#L16

Personally I think `default_version` makes the most sense but I'm open to alternatives.

Reply:
Is there a reason we need to `require` it over setting a default? I am not convinced this needs to be explicitly set. This ends up being another breaking change, making upgrading less seamless.

We already make many assumptions about what the SDK should do... `batch size`, `retries`, `timeouts,` etc. If 99% of the time the answer going forward is going to be `v1.0`, then let's just default it to that and let users who need to opt-out, opt-out explicitly. 

1. Forcing users to set a variable named `default_*` is confusing - if its a default / fallback then we should be opinionated on what that default is.
2. There are no `required` options in `Configuration` - so it is possible to use the Gem without setting up an initializer or configuring the gem. It works out of the box on install. By adding an option that must be set, we break this. Additional documentation needs to be added to explain why this option needs to be set, how to set it. Getting started becomes harder.
3. VAPID has a deprecation plan in place - the default should be `v1.0`.

Reply:
I see your points Patrick and that's the way I originally had it, with a default default of `v1.0`, but it got pushback from @shawnce & @allanrsmith.

Requiring a default forces users to be more explicit about which version they want to hit. Someday if we go to `v2.0`, do we change the default again? It might be confusing if the user upgrades the gem and are suddenly hitting different endpoints with breaking changes.

I'm okay switching it back though if we can get agreement.

Reply:
I'm in favor of switching back to your original way, and default to `v1.0`. If `v2.0` ever becomes standard, we can bump the gem version. It'll shake out nicely that the gem will track REST -  `1.0` of the gem will default to `v1.0 REST`. if `v2.0` becomes norm, we can major version the gem to `2.x`, etc.

Reply:
I've switched it back defaulting to `v1.0`